### PR TITLE
Workbench: User aimed documentation

### PR DESCRIPTION
**Pull request by Pasarus**
**Description of work.**
This is the partially complete workbench documentation aimed at helping users get to grips with the new interface and operate with it.

**To test:**
- Build the user documentation target `docs-html`
- Then find the `/docs/html/index.html` file and then it should have a new section in reference documentation called `workbench` click this
- Here is the workbench documentation this file should be mostly complete
- Any of the files linked from this page are not finished.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DanNixon** [comment]
We have code for generating screenshots in the documentation build (this is where the algorithm screenshots come from), it's worth looking to see how many of the PNGs in this PR can be replaced with this.

**Comment by NickDraper** [comment]
> We have code for generating screenshots in the documentation build (this is where the algorithm screenshots come from), it's worth looking to see how many of the PNGs in this PR can be replaced with this.

@DanNixon Does this work with the workbench?

**Comment by DanNixon** [comment]
@NickDraper I've no idea since I've not worked on workbench. It is worth looking into though as it keeps the repo clean and reduces the need for developers to keep updating screenshots.

It it does not currently work it should be made to work as it will be needed for existing algorithm and interface documentation post MantidPlot.

File: docs/source/workbench/messageswindow.rst [hunk 1]
```diff
@@ -4,15 +4,32 @@
...
 
 .. image:: ../images/Workbench/MessageWindow/MessagesWidget.png
     :align: right
     :scale: 50%
 
-Logging Level
--------------
+All output from running your scripts or Mantid algorithms is logged in
+this window. To minimise the amount of information displayed in this window you
+are provided with 5 logging levels (in ascending order of priority):
```
Comment (Pasarus) [inline docs/source/workbench/messageswindow.rst]: Would it be possible to describe the type of people/content that may be interested/is present at different logging levels?

Comment (hsautessella) [reply docs/source/workbench/messageswindow.rst]: I've added this in commit 5dad4a4. Also `To minimise the amount of information displayed in this window..` isn't the right phrase here, I'll change that in my next commit.

File: docs/source/workbench/messageswindow.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
+.. _WorkbenchMessagesWindow:
+
+===============
+Messages Window
+===============
+
+.. image:: ../images/Workbench/MessageWindow/MessagesWidget.png
```
Comment (NickDraper) [inline docs/source/workbench/messageswindow.rst]: Can we hack out some of the pointless empty height of this image?

File: docs/source/workbench/index.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+**Workbench Features**
+    * :ref:`WorkbenchWorkspaceToolbox` The area in which all workspaces currently loaded into Workbench can be accessed
+      and edited from.
+    * :ref:`WorkbenchAlgorithmToolbox` Here is all of the Algorithms availible to users to run on workspaces loaded into
+      the Workspace Toolbox.
+    * :ref:`WorkbenchScriptWindow` is the Main feature in workbench being right in the middle on startup, it is a python
+      script editor.
+    * :ref:`WorkbenchMessagesWindow` is effectively the results log from MantidPlot, all output from most sources goes
+      here.
+    * :ref:`WorkbenchMainWindowMenu` The new simplified menus in Workbench.
```
Comment (NickDraper) [inline docs/source/workbench/index.rst]: Move this to the top of this list

File: docs/source/workbench/mainwindowmenu.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
+.. _WorkbenchMainWindowMenu:
+
+=================
+Main window menus
+=================
+
+.. image:: ../images/Workbench/MainWindowMenu/MainWindowMenu.png
```
Comment (NickDraper) [inline docs/source/workbench/mainwindowmenu.rst]: Can you see what it looks like is you align some of these to the right?

File: docs/source/workbench/plotstoolbox.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
+.. _WorkbenchPlotsToolbox:
+
+=============
+Plots Toolbox
+=============
+
+.. image:: ../images/Workbench/PlotToolbox/PlotsToolbox.png
```
Comment (NickDraper) [inline docs/source/workbench/plotstoolbox.rst]: Use the image below here, and don't include this one

Comment (NickDraper) [reply docs/source/workbench/plotstoolbox.rst]: Make sure it is big enough to read the text, but not too big

File: docs/source/workbench/ipythonconsole.rst [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+IPython Console
+===============
+
+.. image:: ../images/Workbench/IPythonWidget.png
+
+This is a python interpreter which has access to the Mantid API and can
+interface with the rest of the Workbench. Through the interpreter it is
+possible to interact with Workbench's algorithms, workspaces and plots. It
+allows you to manipulate things much like the script window can, but in an
+interpreter style.
```
Comment (NickDraper) [inline docs/source/workbench/ipythonconsole.rst]: Can we mentino that this executes python automatically after return is pressed, and perhaps include a link to ipython/jupyter?

File: docs/source/workbench/plotwindow.rst [hunk 7]
```diff
@@ -0,0 +1,67 @@
+.. _WorkbenchPlotWindow:
+
+==================
+Workbench Plotting
+==================
+
+.. image:: ../images/Workbench/PlotWindow/PlotWindow.png
```
Comment (NickDraper) [inline docs/source/workbench/plotwindow.rst]: Can we align just this image to the right?

**Review by NickDraper — CHANGES_REQUESTED** [review]
Mostly good, just a few suggested changes

File: docs/source/workbench/algorithmtoolbox.rst [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+.. image:: ../images/Workbench/AlgorithmListWidget/AlgorithmExecuteDialog.png
+
+From this window you can enter your input workspace, the parameters for your
+selected algorithm and your target output workspace.
+
+
+Algorithm Categories
+--------------------
+
+The algorithms are sorted into categories to make navigating easier. Below is a
```
[the remark this answers is not in the document]
Comment (hsautessella) [reply docs/source/workbench/algorithmtoolbox.rst]: As far as I can tell you need to click `Execute` to run an algorithm. Double-clicking doesn't seem to do anything.

**Review by NickDraper — APPROVED** [review]

**Pull request merged**